content/metadata-stats/stats.textile

```diff
@@ -0,0 +1,175 @@
+    ARTStats *thisHour = result.items[0];
+    NSLog(@"Published this hour %lu", thisHour.inbound.all.all.count);
+} error:nil];
+```
+
+```[swift]
+let realtime = ARTRealtime(key: "{{API_KEY}}")
+let query = ARTStatsQuery()
+query.unit = .Hour
+try! realtime.stats(query) { results, error in
+    let thisHour = results!.items[0]
+    print("Published this hour \(thisHour.inbound.all.all.count)")
+}
+```
+
+REST-CODE
+
+```[jsall]
+  var rest = new Ably.Rest('{{API_KEY}}');
+  rest.stats({ unit: 'hour' }, function(err, resultPage) {
+    var thisHour = resultPage.items[0];
+    console.log(thisHour); // => {all: a, inbound: f, outbound: f, …}
+  });
+```
+
+```[ruby]
+  rest = Ably::Rest.new('{{API_KEY}}')
+  result_page = rest.stats(unit: 'hour')
+  this_hour = result_page.items.first
+  puts this_hour # => #<Ably::Models::Stat:…
+```
+
+```[python]
+  rest = AblyRest('{{API_KEY}}')
+  result_page = rest.stats(unit='hour')
+  this_hour = result_page.items[0]
+  print(this_hour) # => ably.types.stats.Stats
+```
+
+```[php]
+  $rest = new Ably\AblyRest('{{API_KEY}}');
+  $resultPage = $rest->stats(array('unit' => 'hour'));
+  $thisHour = $resultPage->items[0];
+  echo('Published this hour ' . $thisHour->inbound->all->all->count);
+```
+
+```[java]
+  AblyRest rest = new AblyRest("{{API_KEY}}");
+  Params options = new Param[]{ new Param("unit", "hour") }
+  PaginatedResult<Stats> results = rest.stats(options);
+  Stats thisHour = results.items[0];
+  System.out.println("Published this hour " + thisHour.inbound.all.all.count);
+```
+
+```[csharp]
+  AblyRest rest = new AblyRest("{{API_KEY}}");
+  StatsRequestParams query = new StatsRequestParams() { Unit = StatsIntervalGranularity.Hour };
+  PaginatedResult<Stats> results = await rest.StatsAsync(query);
+  Stats thisHour = results.Items[0];
+  Console.WriteLine("Published this hour " + thisHour.Inbound.All.All.Count);
+```
+
+```[objc]
+ARTRest *rest = [[ARTRest alloc] initWithKey:@"{{API_KEY}}"];
+ARTStatsQuery *query = [[ARTStatsQuery alloc] init];
+query.unit = ARTStatsGranularityHour;
+[rest stats:query callback:^(ARTPaginatedResult<ARTStats *> *result, ARTErrorInfo *error) {
+    ARTStats *thisHour = result.items[0];
+    NSLog(@"Published this hour %lu", thisHour.inbound.all.all.count);
+} error:nil];
+```
+
+```[swift]
+let rest = ARTRest(key: "{{API_KEY}}")
+let query = ARTStatsQuery()
+query.unit = .Hour
+try! rest.stats(query) { results, error in
+    let thisHour = results!.items[0]
+    print("Published this hour \(thisHour.inbound.all.all.count)")
+}
+```
+
+```[go]
+rest, err := ably.NewRestClient(ably.NewClientOptions("{{API_KEY}}"))
+query := &ably.PaginateParams{
+  ScopeParams: ably.ScopeParams{
+    Unit:  'hour',
+  },
+}
+page, err := rest.Stats(query)
+if err != nil {
+  // Error
+}
+fmt.Println("Published this hour: %d", len(page2.Items[0]))
+```
+
+<span lang="ruby">Note that all examples on this page assume you are running them within an EventMachine reactor.</span>
```

Comment:
What does that mean? Can we explain more?

Reply:
This is Ruby specific and is in multiple places. This has reminded me to log an issue for post-IA for it, as I think we should have this centrally in the setup of an SDK rather than against every code block.